src/Logger.php

```diff
@@ -9,15 +9,18 @@
 use Psr\Log\LoggerTrait;
 use Psr\Log\LogLevel;
 use Throwable;
+use Yiisoft\Log\Message\CategoryFilter;
 
 use function array_filter;
+use function count;
 use function debug_backtrace;
 use function gettype;
 use function get_class;
 use function implode;
 use function in_array;
 use function is_string;
 use function memory_get_usage;
+use function microtime;
```

Comment:
`getTarget()` isn't currently used.

1. Should be expose it?
2. Should we assume targets are named?

Reply:
Should we allow dynamically adding targets via `addTarget()`?

Reply:
`getFlushInterval()` and `getTraceLevel()` aren't used except tests. Should we expose these?

The class overall isn't immutable. Should it be?